src/digest.rs

```diff
@@ -0,0 +1,86 @@
+use crate::raw;
+use std::os::raw::c_char;
+
+/// `Digest` is a high-level rust interface to the Valkey module C API
+/// abstracting away the raw C ffi calls.
+pub struct Digest {
+    pub dig: *mut raw::RedisModuleDigest,
+}
+
+impl Digest {
```

Comment:
Nit: For all the unsafe functions where we are unpacking the Module API into an optional function, we can use a simpler syntax with `expect`. Example:
```
unsafe {
    raw::RedisModule_GetKeyNameFromDigest
        .expect("RedisModule_GetKeyNameFromDigest is not available.")(self.dig)
}
```

Reply:
Great feedback - [updated](https://github.com/valkey-io/valkeymodule-rs/pull/102/commits/ff96f1b2583c3ec2896e405b01bc8854aa241316)